pantheon/src/test/java/tech/pegasys/pantheon/cli/operator/OperatorSubCommandTest.java

```diff
@@ -0,0 +1,217 @@
+    parseCommand(OPERATOR_SUBCOMMAND_NAME);
+    assertThat(commandOutput.toString()).startsWith(EXPECTED_OPERATOR_USAGE);
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
+  @Test
+  public void callingOperatorCommandHelpMustDisplayUsage() {
+    parseCommand(OPERATOR_SUBCOMMAND_NAME, "--help");
+    assertThat(commandOutput.toString()).startsWith(EXPECTED_OPERATOR_USAGE);
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
+  @Test
+  public void generateBlockchainConfigMustGenerateKeysWhenGenerateIsTrue() throws IOException {
+    generateBlockchainConfig(
+        cmd(),
+        "/operator/config_generate_keys.json",
+        tmpOutputDirectoryPath,
+        "genesis.json",
+        true,
+        asList("key.pub", "key.priv"));
+  }
+
+  @Test
+  public void generateBlockchainConfigMustImportKeysWhenGenerateIsFalse() throws IOException {
+    generateBlockchainConfig(
+        cmd(),
+        "/operator/config_import_keys.json",
+        tmpOutputDirectoryPath,
+        "genesis.json",
+        false,
+        singletonList("key.pub"));
+  }
+
+  @Test
+  public void genesisFileNameShouldBeEqualToOption() throws IOException {
+    generateBlockchainConfig(
+        cmd("--genesis-file-name", "option.json"),
+        "/operator/config_generate_keys.json",
+        tmpOutputDirectoryPath,
+        "option.json",
+        true,
+        asList("key.pub", "key.priv"));
+  }
+
+  @Test
+  public void publicKeyFileNameShouldBeEqualToOption() throws IOException {
+    generateBlockchainConfig(
+        cmd("--public-key-file-name", "pub.test"),
+        "/operator/config_generate_keys.json",
+        tmpOutputDirectoryPath,
+        "genesis.json",
+        true,
+        asList("pub.test", "key.priv"));
+  }
+
+  @Test
+  public void privateKeyFileNameShouldBeEqualToOption() throws IOException {
+    generateBlockchainConfig(
+        cmd("--private-key-file-name", "priv.test"),
+        "/operator/config_generate_keys.json",
+        tmpOutputDirectoryPath,
+        "genesis.json",
+        true,
+        asList("key.pub", "priv.test"));
+  }
+
+  private void generateBlockchainConfig(
+      final Cmd cmd,
+      final String configFile,
+      final Path outputDirectoryPath,
+      final String genesisFileName,
+      final boolean generate,
+      final Collection<String> expectedKeyFiles) {
+    final URL configFilePath = this.getClass().getResource(configFile);
+    final DocumentContext configDocumentContext = JsonPath.parse(contentOf(configFilePath, UTF_8));
+    parseCommand(
+        cmd(
+                OPERATOR_SUBCOMMAND_NAME,
+                GENERATE_BLOCKCHAIN_CONFIG_SUBCOMMAND_NAME,
+                "--config-file",
+                configFilePath.getPath(),
+                "--to",
+                outputDirectoryPath.toString())
+            .args(cmd.argsArray())
+            .argsArray());
+    assertThat(commandErrorOutput.toString()).isEmpty();
```

Comment:
(optional) This method name `generateBlockchainConfig` doesn't indicate checks are being performed, so these asserts are a little unexpected.  What if you break up this method into 2 smaller methods (maybe `runCommand` and then `validateCmdOutput`)?

Reply:
method has been renamed to give more context